test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -368,34 +369,61 @@ private String formatStatusCodeMessage(ClientYamlTestResponse restTestResponse,
                 not(equalTo(409)))));
     }
 
-    private static NodeSelector buildNodeSelector(XContentLocation location, String name, String value) {
+    private static NodeSelector buildNodeSelector(String name, XContentParser parser) throws IOException {
         switch (name) {
+        case "attribute":
+            return parseAttributeValuesSelector(parser);
         case "version":
-            Version[] range = SkipSection.parseVersionRange(value);
-            return new NodeSelector() {
-                @Override
-                public void select(Iterable<Node> nodes) {
-                    for (Iterator<Node> itr = nodes.iterator(); itr.hasNext();) {
-                        Node node = itr.next();
-                        if (node.getVersion() == null) {
-                            throw new IllegalStateException("expected [version] metadata to be set but got "
-                                    + node);
-                        }
-                        Version version = Version.fromString(node.getVersion());
-                        if (false == (version.onOrAfter(range[0]) && version.onOrBefore(range[1]))) {
-                            itr.remove();
-                        }
+            return parseVersionSelector(parser);
+        default:
+            throw new XContentParseException(parser.getTokenLocation(), "unknown node_selector [" + name + "]");
+        }
+    }
+
+    private static NodeSelector parseAttributeValuesSelector(XContentParser parser) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new XContentParseException(parser.getTokenLocation(), "expected START_OBJECT");
+        }
+        String key = null;
+        XContentParser.Token token;
+        NodeSelector result = NodeSelector.ANY;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                key = parser.currentName();
+            } else {
+                NodeSelector newSelector = new HasAttributeNodeSelector(key, parser.text());
+                parser.skipChildren();
```

Comment:
do we need to skip children here?

Reply:
Let me dig on these. I *think* they are nice if you pass in weird stuff. Because we don't check the type and just call `text`. I'll see.